canonical-web-and-design/maas-ui
# refactor(ui): replace Redirect with useNavigate

## Done

- Replace Redirect with useNavigate from React Router v6.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that it redirects to the scripts list after uploading a script in settings.
- Check that visiting the instances tab `/machine/:id/instances` for a machine without any instances redirects back to the summary.
- Click "KVM" in the header and you should be taken to /kvm/lxd.
- Check that KVM redirects work (try the QA steps here: https://github.com/canonical-web-and-design/maas-ui/pull/3179).
- Visit the /intro after having completed it and you should be redirected to the dashboard.
- Try saving some add forms and you should be redirected back to lists.

## Fixes

Fixes: #3962.
